File: OnlineRepairBooking/frontend/src/api/service.js
import request from '@/utils/request'

export const getCategories = () => {
  return request({
    url: '/services/categories',
    method: 'get'
  })
}

export const getServices = (params) => {
  return request({
    url: '/services',
    method: 'get',
    params
  })
}

export const getServiceDetail = (id) => {
  return request({
    url: `/services/${id}`,
    method: 'get'
  })
}

export const getTimeSlots = (params) => {
  return request({
    url: '/services/time-slots',
    method: 'get',
    params
  })
}

File: OnlineRepairBooking/backend/internal/routes/service.go
package routes

import (
	"online-repair-booking/internal/handlers"
	"online-repair-booking/pkg/database"

	"github.com/labstack/echo/v4"
)

func RegisterServiceRoutes(g *echo.Group) {
	db := database.MySQL
	serviceHandler := handlers.NewServiceHandler(db)

	serviceGroup := g.Group("/services")
	{
		serviceGroup.GET("/categories", serviceHandler.GetCategoryList)
		serviceGroup.GET("", serviceHandler.GetServiceList)
		serviceGroup.GET("/:id", serviceHandler.GetServiceDetail)
		serviceGroup.GET("/time-slots", serviceHandler.GetTimeSlots)
	}
}
